src/index.js

```diff
@@ -338,3 +282,84 @@ let Rosnodejs = {
 Rosnodejs.ActionServer = ActionServer;
 
 module.exports = Rosnodejs;
+
+//------------------------------------------------------------------
+// Local Helper Functions
+//------------------------------------------------------------------
+
+/**
+ * @private
+ * Helper function to see if the master is available and able to accept
+ * connections.
+ * @param {number} timeout time in ms between connection attempts
+ * @param {number} maxTimeout maximum time in ms to retry before timing out.
+ * A negative number will make it retry forever. 0 will only make one attempt
+ * before timing out.
+ */
+function _checkMasterHelper(timeout=100, maxTimeout=-1) {
+  let startTime = Date.now();
+  const localHelper = (resolve,reject) => {
+    pingMasterTimeout = setTimeout(() => {
+      // also check that the slave api server is set up
+      if (!rosNode.slaveApiSetupComplete()) {
+        if (Date.now() - startTime >= maxTimeout && !(maxTimeout < 0) ) {
+          log.error(`Unable to register with master node [${rosNode.getRosMasterUri()}]: unable to set up slave API Server. Stopping...`);
+          reject(Error('Unable to setup slave API server.'));
+          return;
+        }
+        localHelper(resolve, reject);
+        return;
+      }
+      rosNode.getMasterUri({ maxAttempts: 1 })
+      .then(() => {
+        log.infoOnce(`Connected to master at ${rosNode.getRosMasterUri()}!`);
+        pingMasterTimeout = null;
+        resolve();
+      })
+      .catch((err, resp) => {
+        if (Date.now() - startTime >= maxTimeout && !(maxTimeout < 0) ){
+          log.error(`Timed out before registering with master node [${rosNode.getRosMasterUri()}]: master may not be running yet.`);
+          reject(Error('Registration with master timed out.'));
+          return;
+        } else {
+          log.warnThrottle(60000, `Unable to register with master node [${rosNode.getRosMasterUri()}]: master may not be running yet. Will keep trying.`);
+          localHelper(resolve, reject);
+        }
+      });
+    }, timeout);
+  };
+
+  return new Promise((resolve, reject) => {
+    localHelper(resolve,reject);
+  });
+}
+
+function _resolveNodeName(nodeName, remappings, options) {
+  let namespace = remappings['__ns'] || process.env.ROS_NAMESPACE || '';
+  namespace = names.clean(namespace);
+  if (namespace.length === 0 || !namespace.startsWith('/')) {
+    namespace = '/' + namespace;
```

Comment:
Be consistent with string literal usage. E.g since template is widely used in this repo, should use:
```
`/${namespace}`
```